src/components/definition-list/definition-list.test.js

```diff
@@ -0,0 +1,208 @@
+        <Box width="90%">
+          <Typography color="rgba(0,0,0,0.55)" variant="segment-subheader-alt">
+            Segment Header
+          </Typography>
+          <Hr ml={0} mt={2} />
+        </Box>
+        <Box mb={3} display="flex">
+          <Box flexGrow="1">
+            <Dl dtTextAlign="left" asSingleColumn>
+              <Dt>First</Dt>
+              <Dd>Description 1</Dd>
+              <Dt>Second</Dt>
+              <Dd>Description 2</Dd>
+              <Dt>Third</Dt>
+              <Dd>Description</Dd>
+            </Dl>
+          </Box>
+        </Box>
+      </Box>
+    </div>
+  );
+};
+
+context("Testing Definition List component", () => {
+  describe("should render Definition List component", () => {
+    it.each(alignValue)(
+      "should verify Definition List text is %s aligned",
+      (align) => {
+        CypressMountWithProviders(
+          <DLComponent dtTextAlign={align} ddTextAlign="right" />
+        );
+
+        getDataElementByValue("dt").should("have.css", "text-align", align);
+      }
+    );
+
+    it.each(alignValue)(
+      "should verify Definition List definition is %s aligned",
+      (align) => {
+        CypressMountWithProviders(<DLComponent ddTextAlign={align} />);
+
+        getDataElementByValue("dd").should("have.css", "text-align", align);
+      }
+    );
+
+    it.each(widths)(
+      "should verify text width is %spx and definition width is %spx, %s%/%s% of the Definition List width",
+      (dtPixels, ddPixels, dtPercent) => {
+        CypressMountWithProviders(<DLComponent w={dtPercent} />);
+
+        getDataElementByValue("dl")
+          .children()
+          .should("have.css", "width", `${dtPixels}px`);
+        getDataElementByValue("dl")
+          .children()
+          .eq(1)
+          .should("have.css", "width", `${ddPixels}px`);
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should check Definition List text is %s when children prop is set to %s",
+      (text) => {
+        CypressMountWithProviders(
+          <Dl>
+            <Dt>{text}</Dt>
+            <Dd>Definition</Dd>
+          </Dl>
+        );
+
+        getDataElementByValue("dt").should("have.text", text);
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should check Definition List definition is %s when children prop is set to %s",
+      (definition) => {
+        CypressMountWithProviders(
+          <Dl>
+            <Dt>Text</Dt>
+            <Dd data-element="dd">{definition}</Dd>
+          </Dl>
+        );
+
+        getDataElementByValue("dd").should("have.text", definition);
+      }
+    );
+
+    it("should verify Definition List text is displayed as a single column", () => {
+      CypressMountWithProviders(
+        <DLComponent dtTextAlign="left" asSingleColumn />
+      );
+
+      getDataElementByValue("dt")
+        .should("have.css", "text-align", "left")
+        .and("have.css", "width", "1350px");
```

Comment:
should here be `width` with the same value in both assertions ?

Reply:
Yes, the width should be the same in both assertions. Normally the text and definition are displayed on the same line, so the default width is 675px each. This test is for the single column prop, which displays the text and definition on separate lines - verifying that the width of each is 1350px shows that they are displayed as a single column.